File: algorithms/2391.minimum-amount-of-time-to-collect-garbage/solution.rb
# @param {String[]} garbage
# @param {Integer[]} travel
# @return {Integer}
def garbage_collection(garbage, travel)
  t_metal = 0
  t_paper = 0
  t_glass = 0

  total_time = 0
  travel.each_with_index do |t, i|
    total_time += t
    t_metal = total_time if garbage[i + 1].include?('M')
    t_paper = total_time if garbage[i + 1].include?('P')
    t_glass = total_time if garbage[i + 1].include?('G')
  end

  garbage.map(&:length).sum + t_metal + t_paper + t_glass
end

File: algorithms/2391.minimum-amount-of-time-to-collect-garbage/solution.rs
impl Solution {
    pub fn garbage_collection(garbage: Vec<String>, travel: Vec<i32>) -> i32 {
        let (mut t_metal, mut t_paper, mut t_glass) = (0, 0, 0);

        let mut total_time = 0i32;
        for (i, t) in travel.iter().enumerate() {
            total_time += t;
            if garbage[i + 1].contains('M') {
                t_metal = total_time;
            }
            if garbage[i + 1].contains('P') {
                t_paper = total_time;
            }
            if garbage[i + 1].contains('G') {
                t_glass = total_time;
            }
        }

        garbage.join("").len() as i32 + t_metal + t_paper + t_glass
    }
}
